### PR TITLE
Fix single action banner to ensure button alignment

### LINKED ISSUE #39574: MaterialBanner button alignment is incorrect
When using a `MaterialBanner` with only one action the button appears on the same line as the content. This is intended.

However, the button needs to have space between it and the content. See the following screenshot. A spacer should be added in this scenario.

Simulator Screen Shot - iPhone 8 Plus - 2019-08-30 at 13 12 37

### PR DESCRIPTION
**Pull request by rami-a**
## Description

Fixes a small layout issue with a single action `MaterialBanner` to ensure it is aligned to the trailing edge of the banner.

| Before  | After |
| ------------- | ------------- |
|   |  |

## Related Issues

Fixes #39574

## Tests

I added the following tests:

* A test that ensures the placement of the buttons is correct.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by willlarche — APPROVED** [review]

**Review by clocksmith — APPROVED** [review]

File: packages/flutter/test/material/banner_test.dart [hunk 1]
```diff
@@ -100,6 +100,28 @@ void main() {
     expect(contentBottomLeft.dx, lessThan(actionsTopRight.dx));
   });
 
+  // Regression for https://github.com/flutter/flutter/issues/39574
```
Comment (HansMuller) [inline packages/flutter/test/material/banner_test.dart]: Regression test for...

File: packages/flutter/test/material/banner_test.dart [hunk 2]
```diff
@@ -100,6 +100,28 @@ void main() {
...
+  testWidgets('Single action laid out beside content but aligned to the trailing edge', (WidgetTester tester) async {
```
Comment (HansMuller) [inline packages/flutter/test/material/banner_test.dart]: Would be helpful to test the RTL case as well.

**Review by HansMuller — APPROVED** [review]
LGTM modulo some small stuff. Maybe include before/after screenshots in the issue's description, to make it easier to understand what changed.

File: packages/flutter/test/material/banner_test.dart (hunk 2 above)
Comment (rami-a) [reply packages/flutter/test/material/banner_test.dart]: Good idea, done

File: packages/flutter/test/material/banner_test.dart (hunk 1 above)
Comment (rami-a) [reply packages/flutter/test/material/banner_test.dart]: Done

**Pull request merged**